src/vs/workbench/services/notebook/common/notebookDocumentService.ts

```diff
@@ -66,6 +66,35 @@ export function generateMetadataUri(notebook: URI): URI {
 	return notebook.with({ scheme: Schemas.vscodeNotebookMetadata, fragment });
 }
 
+export function parseCellOutputUri(uri: URI): { notebook: URI; openIn: string; outputId?: string; cellFragment?: string; outputIndex?: number } | undefined {
+	if (uri.scheme !== Schemas.vscodeNotebookCellOutput) {
+		return;
+	}
+
+	const params = new URLSearchParams(uri.query);
+	const openIn = params.get('openIn');
+	if (!openIn) {
+		return;
+	}
+	const outputId = params.get('outputId') ?? undefined;
+	const cellFragment = uri.fragment ?? undefined;
+	const outputIndex = params.get('outputIndex') ? parseInt(params.get('outputIndex') || '', 10) : undefined;
```

Comment:
Same code in `notebookCommon.ts`
Can we not reuse that same function?

Reply:
had to duplicate since it is used in labels which is too low level to reference notebookCommon from my understanding

Reply:
Lets get this PR out for TPI, 
please can you file an issue for debt week to fix this, else we'd end up with issues if we ever make a change in one of the two places and not the other (i.e. usual problems that arise from duplicating)